live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/domain/event/IngestNewNodeEventHandlerTest.java

```diff
@@ -26,89 +26,79 @@
 
 package org.alfresco.hxi_connector.live_ingester.domain.event;
 
-import static org.alfresco.hxi_connector.live_ingester.domain.model.out.PredefinedNodeProperty.*;
-import static org.alfresco.hxi_connector.live_ingester.util.TestUtils.assertContainsSameElements;
-import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.Mockito.mock;
+import static org.mockito.BDDMockito.given;
 import static org.mockito.Mockito.verify;
-
-import java.util.Set;
+import static org.mockito.Mockito.verifyNoInteractions;
 
 import org.alfresco.hxi_connector.live_ingester.domain.model.in.IngestNewNodeEvent;
 import org.alfresco.hxi_connector.live_ingester.domain.model.in.Node;
 import org.alfresco.hxi_connector.live_ingester.domain.model.out.EventPublisher;
-import org.alfresco.hxi_connector.live_ingester.domain.model.out.NodeProperty;
 import org.alfresco.hxi_connector.live_ingester.domain.model.out.event.UpdateNodeMetadataEvent;
+import org.alfresco.hxi_connector.live_ingester.domain.model.transform.request.TransformRequest;
+import org.alfresco.hxi_connector.live_ingester.domain.model.transform.request.TransformRequester;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.ArgumentCaptor;
-import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 
 @ExtendWith(MockitoExtension.class)
 class IngestNewNodeEventHandlerTest
 {
-    private static final long EVENT_TIMESTAMP = 1_690_000_000_100L;
-    private static final String NODE_ID = "0fe2919a-e0a6-4033-8d35-168a16cf33fc";
-    private static final String NODE_NAME = "test-name";
-    private static final String NODE_PRIMARY_ASSOC_Q_NAME = "cm:test-name";
-    private static final String NODE_TYPE = "cm:folder";
-    private static final String NODE_CREATED_BY_USER_WITH_ID = "admin";
-    private static final String NODE_MODIFIED_BY_USER_WITH_ID = "admin";
-    private static final Set<String> NODE_ASPECT_NAMES = Set.of(
-        "cm:titled",
-        "cm:auditable");
-    private static final boolean NODE_IS_FOLDER = true;
-    private static final boolean NODE_IS_FILE = false;
-    private static final long NODE_CREATED_AT = 1_690_000_000_050L;
-    private static final NodeProperty<String> NODE_TITLE = new NodeProperty<>("cm:title", "some title");
-    private static final Set<NodeProperty<?>> NODE_PROPERTIES = Set.of(NODE_TITLE);
-    @Captor
-    ArgumentCaptor<UpdateNodeMetadataEvent> updateNodeMetadataEventCaptor;
-    private final EventPublisher eventPublisher = mock();
-    private final IngestNewNodeEventHandler ingestNewNodeEventHandler = new IngestNewNodeEventHandler(eventPublisher);
+    @Mock
+    IngestNewNodeEvent ingestNewNodeEvent;
+    @Mock
+    Node node;
+    @Mock
+    UpdateNodeMetadataEvent updateNodeMetadataEvent;
+    @Mock
+    TransformRequest transformRequest;
```

Comment:
The way I was thinking about these unit test is:
* Static field declarations contain CONSTANTS which are mainly passed around
* Field declarations contains a list of mocks and one real class
* `setUp` contains background information that applies to every test (in this class it's one line) wiring up mocks
* Each test method contains the important parts of the test that you have to read to understand what it's doing

For this test we could instead do something like you suggested:
```
    IngestNewNodeEvent mockedIngestNewNodeEvent(String contentMimeType)
    {
        Node node = mock();
        given(node.contentMimeType()).willReturn(contentMimeType);

        IngestNewNodeEvent ingestNewNodeEvent = mock();
        given(ingestNewNodeEvent.node()).willReturn(node);
        return ingestNewNodeEvent;
    }
```
This removes a couple of lines from what you wrote, but I think it's still harder to follow than what I've done.

Reply:
the thing is that you don't really want to hide everything under some method - in test you want to show that:

when you will receive event with node with some content type
then event handler should update metadata properties and request content transformation

so you can also do something like this:
```
@Test
void nodeWithContent_PublishMetadataAndRequestTransform()
{
    // given
    Node node = mockedNode("application/msword");
    IngestNewNodeEvent ingestNewNodeEvent = mockedIngestNewNodeEvent(node);

    // when
    ingestNewNodeEventHandler.handle(ingestNewNodeEvent);

    // then
    verify(eventPublisher).publishMessage(any());
    verify(transformRequester).requestTransform(any());
    }
```
but you want to see that you have node and you want to put this node into an event

Reply:
As an outsider, I must admit that it was easier for me to read Paweł's proposition. I would only change method names, e.g. from `mockedNode()` to `mockNode()`, because the original one is not `mocked()`, it's `mock()`.